modules/drivers/bigquery-cloud-sdk/src/metabase/driver/bigquery_cloud_sdk.clj

```diff
@@ -268,9 +278,10 @@
     (try
       (thunk)
       (catch Throwable e
-        (if-not (error-type/client-error? (:type (u/all-ex-data e)))
-          (thunk)
-          (throw e))))))
+        (let [ex-data (u/all-ex-data e)]
+          (if (or (:retryable? e) (not (error-type/client-error? (:type ex-data))))
+            (thunk)
```

Comment:
do we need any state here to only retry `n` times, put a backoff strategy into retrying, etc?

Reply:
Good point.  Looking more deeply now, it appears the library itself is doing this to some degree.  Check out this block (the default behavior they have, which is actually defined in a separate source file is looking at the retryable flag).

The #17959 PR is doing some related stuff so let's see whether it even makes sense to keep our own retry logic after everything gets sorted.